src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.AbstractIndenter.cs

```diff
@@ -11,30 +11,32 @@
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
-using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.Implementation.SmartIndent
+namespace Microsoft.CodeAnalysis.Indentation
 {
     internal abstract partial class AbstractIndentationService<TSyntaxRoot>
     {
         internal abstract class AbstractIndenter
         {
             public readonly OptionSet OptionSet;
-            protected readonly TextLine LineToBeIndented;
+            public readonly TextLine LineToBeIndented;
             protected readonly int TabSize;
             protected readonly CancellationToken CancellationToken;
 
+            protected readonly Document Document;
             protected readonly TSyntaxRoot Root;
             protected readonly SyntaxTree Tree;
             protected readonly IEnumerable<AbstractFormattingRule> Rules;
             protected readonly BottomUpBaseIndentationFinder Finder;
 
             private static readonly Func<SyntaxToken, bool> s_tokenHasDirective = tk => tk.ContainsDirectives &&
                                                   (tk.LeadingTrivia.Any(tr => tr.IsDirective) || tk.TrailingTrivia.Any(tr => tr.IsDirective));
+
             private readonly ISyntaxFactsService _syntaxFacts;
 
             public AbstractIndenter(
-                ISyntaxFactsService syntaxFacts,
+                Document document,
```

Comment:
Syntaxdocument? so that we don't get confused or need to add assert saying document and tree is from same snapshot?

Reply:
done.  thanks!